test/unit/client-daemonset.bats

```diff
@@ -86,6 +86,32 @@ load _helpers
   [ "${actual}" = "true" ]
 }
 
+@test "client/DaemonSet: retry join gets populated when client.join is set" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/client-daemonset.yaml  \
+      --set 'server.enabled=false' \
+      --set 'externalServers.enabled=true' \
+      --set 'client.join[0]=1.1.1.1' \
+      . | tee /dev/stderr |
+      yq -r '.spec.template.spec.containers[0].command | any(contains("-retry-join=\"1.1.1.1\""))' | tee /dev/stderr)
+
+  [ "${actual}" = "true" ]
+}
+
+@test "client/DaemonSet: retry join gets populated when externalServers.hosts is set" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/client-daemonset.yaml  \
+      --set 'server.enabled=false' \
+      --set 'externalServers.enabled=true' \
+      --set 'externalServers.hosts[0]=1.1.1.1' \
+      . | tee /dev/stderr |
+      yq -r '.spec.template.spec.containers[0].command | any(contains("-retry-join=\"1.1.1.1:8301\""))' | tee /dev/stderr)
+
+  [ "${actual}" = "true" ]
+}
+
```

Comment:
Can you also test:
* setting `externalServers.serfLANPort`
* setting multiple join/host urls